### DISCUSSION TITLE
Is there a way to determine from the command line if Valent is connected to my phone?

### DISCUSSION THREAD

**Discussion by joekm** [andyholmes/valent#610] [Help] [answered]
I'm looking to write a simple Waybar module that can launch Valent and show a different icon depending if it's connected to my phone or not.  Just wondering if there's a simple query I could run?

**Comment by andyholmes** [comment, marked as answer]
The only way to get such information is through D-Bus, so you could query the D-Bus object manager:

```sh
gdbus call --session \
           --dest "ca.andyholmes.Valent" \
           --object-path "/ca/andyholmes/Valent" \
           --method "org.freedesktop.DBus.ObjectManager.GetManagedObjects"
```

Or if waybar plugins have some support for D-Bus there may be some more convenient methods you can use.

**Reply by joekm** [reply]
It's something I can play around with, thanks :).
